crates/polars-plan/src/logical_plan/optimizer/count_star.rs

```diff
@@ -0,0 +1,85 @@
+use std::path::PathBuf;
+
+use super::*;
+
+pub(super) struct CountStar;
+
+impl CountStar {
+    pub(super) fn new() -> Self {
+        Self
+    }
+}
+
+impl OptimizationRule for CountStar {
+    // Replace select count(*) from datasource with specialized map function
+    fn optimize_plan(
+        &mut self,
+        lp_arena: &mut Arena<ALogicalPlan>,
+        _expr_arena: &mut Arena<AExpr>,
+        node: Node,
+    ) -> Option<ALogicalPlan> {
+        let mut paths = Vec::new();
+        visit_logical_plan_for_scan_paths(&mut paths, node, lp_arena).map(|scan_type| {
+            let placeholder = ALogicalPlan::DataFrameScan {
```

Comment:
Maybe a comment here about why we need a placeholder. As the node expects an input.